src/util/RouterUtil.js

```diff
@@ -118,7 +118,8 @@ function (Okta, Util, OAuth2Util, Enums, BrowserFeatures, Errors, ErrorCodes) {
 
       var successData = {
         user: res._embedded.user,
-        type: res.type
+        type: res.type,
+        relayState: res.relayState
```

Comment:
is this always defined? We are adding it to all success callbacks, so do we know if all SUCCESS api responses have this value? if not all have the value, are we ok with sending undefined/empty string in the response?
A consumer of the widget will see in the data we pass to the success function the usual data + an relayState (empty or with a value). So just want to be sure that is the intention

Reply:
main-loginpage.js check is:

if (featureRequiresStepUp && res.stepUp) {
      res.stepUp.finish();
    } else if(res.idpDiscovery) {
      res.idpDiscovery.redirectToIdp(config.fromUri);
    } else if (res.relayState) {
      res.session.setCookieAndRedirect(res.relayState);
    } else {
      // Normal flow: Navigate to the sessionCookieRedirect endpoint
      res.session.setCookieAndRedirect(config.redirectUri);
    }

this will work only if relayState is populated.

Reply:
@uvartak-okta @mauriciocastillosilva-okta 
Would it be best to do a check to see the if relayState is present and only then add to the successData object ?

Reply:
``` 
var successData = {
        user: res._embedded.user,
        type: res.type
};
if (res.relayState) {
   successData.relayState = res.relayState;
}

```